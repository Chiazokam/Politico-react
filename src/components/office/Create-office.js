import React, {Component} from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import { Container, FormField, Button } from '../global';
import '../../styles/create-party/create-party.scss';
import { createOffice, getOffices } from '../../actions';
 
class CreateOffice extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      type: ''
    };
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const office = {
      name: this.state.name,
      type: this.state.type
    }
   const { offices, createOffice } = this.props;
   createOffice(office);
   offices();
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value});
  }

  clearField = (e) => {
    this.setState({ errors: '' })
  }

  render() {
    const { errors, redirect, submit } = this.props;
    if (redirect) return <Redirect to='/offices'/>;
    return (
      <div id='create-office'>
        <Container>
          <p className='form-text'>Create Party</p>
          <p className='form-info'>All fields are compulsory</p>
            <form onSubmit={this.handleSubmit}>
              <div className='error'>{errors.message}</div>
              <FormField 
                className='form-field'
                type='text'
                error={errors.name}
                name='name'
                onFocus={this.clearField}
                onChange={this.handleChange}
                value={this.state.name}
                placeholder='Party Name'
              />

              <div className='form-field'>
                <div id='error-type'></div>
                <select name='type' onChange={this.handleChange}>
                  <option>Select Office Type</option>
                  <option value='federal'>Federal</option>
                  <option value='legislative'>Legislative</option>
                  <option value='state'>State</option>
                  <option value='local government'>Local Government</option>
                </select>
              </div>

              <Button
                type='submit'
                className='btn btn-colored btn-signin btn-dark'>
                { submit && <FontAwesomeIcon 
                  icon={ faSpinner }
                  spin
                /> }
                &nbsp; Submit
              </Button>
            </form>
        </Container>
      </div>
    )
  }
}

const mapDispatchToProps = {
  offices: getOffices,
  createOffice,
};

const mapStateToProps = state => ({
  office: state.office.office,
  errors: state.office.errors,
  redirect: state.office.redirect,
  submit: state.office.submit
});

export default connect(mapStateToProps, mapDispatchToProps)(CreateOffice);
