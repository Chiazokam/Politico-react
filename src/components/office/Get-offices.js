import React, { Component } from 'react';
import { connect } from 'react-redux';
import { OfficeCard } from '../global';
import { getOffices } from '../../actions';

class GetOffices extends Component {
  componentDidMount() {
    this.getOffices();
  }

  getOffices = () => {
    const { dispatch } = this.props;
    dispatch(getOffices());
  }

  render() {
    const { getOffices } = this.props;
    return (
      <React.Fragment>
        {getOffices.length === 0 ? 
        <div>
          <div className="card-party title-header">
            <div className="party-main-admin party-main-parties admin-view-main">No Offices Found</div>
          </div>
        </div> :
        <div>
          <div className="card-party title-header">
            <div className="office-main">Office</div> 
            <div className="office-vote office-vote-flex">
            <div className="candidate-party">Office Type</div>
            </div> 
        </div> 
        <div className="card-party title-mobile">
            <div className="party-main admin-view-main">Offices</div>
        </div>
        </div> }

        { getOffices.map(office => 
          <OfficeCard
            key={Math.random(Date.now())}
            name={office.name}
            type={office.type}
        />
        )}
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => ({
  getOffices: state.office.getOffices,
});

export default connect(mapStateToProps)(GetOffices);
