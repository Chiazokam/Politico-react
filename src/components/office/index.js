import { CreateOffice } from './Create-office';
import { GetOffices } from './Get-offices';

export { CreateOffice, GetOffices };
