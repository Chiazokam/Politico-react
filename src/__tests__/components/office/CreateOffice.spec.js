import React from 'react';
import "@babel/polyfill";
import { shallow } from 'enzyme';
import { CreateOfficeUnit } from '../../../components/office/Create-office';

describe('<Create Office />', () => {
  it('renders correctly', () => {
    const props = {
      errors: {
        message: 'Wrong input',
        office: 'Chief Judge'
      },
      offices: jest.fn()
    };
    const tree = shallow(<CreateOfficeUnit {...props} />);
    expect(tree).toMatchSnapshot(); 
  });
});
