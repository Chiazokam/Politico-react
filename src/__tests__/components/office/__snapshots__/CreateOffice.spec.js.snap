// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create Office /> renders correctly 1`] = `ShallowWrapper {}`;
